### DISCUSSION TITLE
How to compute the QueriesResultLimit?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#661] [Q&A] [answered]
I get the below error message during visualization:
![image](https://user-images.githubusercontent.com/35210495/182340021-ef1d276c-11fb-45ef-a31b-eeb6b69fb352.png)

How do we compute the limit size that should be set? I don't quite get the below:
```Please increase the IQueriesResultLimit context value accordingly.[points size before contribution = 1000000, contribution size = 1, max limit = 1000000]```

Also, instead of applying the context for the whole cube (https://docs.atoti.io/latest/lib/atoti/atoti.cube.html#atoti.Cube.shared_context), can we apply it to the specific visualization?

**Comment by bloussou** [comment, marked as answer]
Hello,
For [cube.query()](https://docs.atoti.io/latest/lib/atoti/atoti.cube.html#atoti.Cube.query) and [session.query_mdx()](https://docs.atoti.io/latest/lib/atoti-query/atoti_query.query_session.html#atoti_query.QuerySession.query_mdx) you have a `context` parameter.

I'm looking for a way to have it in notebooks visualizations

**Reply by tibdex** [reply]
https://github.com/atoti/atoti/issues/195#issuecomment-744371298 mentions how to add context values to a widget from JupyterLab. From the expected shape of the state you can enable the [`ADVANCED_APP_EXTENSION`](https://docs.atoti.io/latest/lib/atoti-plus/atoti_plus.app_extension.html#atoti_plus.app_extension.ADVANCED_APP_EXTENSION) and then use the context value editor in the app to change to context values of a widget and see how it impacts its state when saving it.

**Reply by HuifangYeo** [reply]
The problem can indeed be overcome by configuring `shared_context` but this is at the cube level.
```
# cube.shared_context["queriesResultLimit.intermediateLimit"] = 4000000
# cube.shared_context["queriesResultLimit.transientLimit"] = 40000000
```
Enabling the `ADVANCED_APP_EXTENSION` leads to another interesting problem. I configured the `queriesResultLimit.intermediateLimit` up to even `10000000` (higher than the shared_context value which worked), the query simply won't run.
![image](https://user-images.githubusercontent.com/35210495/182513418-72ff46e8-fd65-498a-892e-3af25c50d240.png)

Below shows how we can modify the cell metadata in JupyterLab previously:
![image](https://user-images.githubusercontent.com/35210495/182513820-d07c677d-a8a7-4a89-9827-2cf6b58cbc49.png)

Referring to the state editor, I have added the configuration at the same level as the `query`. It seems to be taking effect based on the changes in the error message. However, similar to the UI, I'm not sure what is the value I need to set in order for it to work:
![image](https://user-images.githubusercontent.com/35210495/182514131-9795182e-c07e-476c-bc5e-8331e8facebc.png)

```
{
    "atoti": {
        "widget": {
            "filters": [
				...
            ],
            "queryContext": [
                {
                    "key": "queriesResultLimit.intermediateLimit",
                    "value": "5000000"
                },
                {
                    "key": "queriesResultLimit.transientLimit",
                    "value": "5000000"
                }
            ],
            "mapping": {
				...
            },
            "query": {
				...
            },
            "serverKey": "default",
            "widgetKey": "pivot-table"
        }
    },
    "tags": []
}
```

**Reply by bloussou** [reply]
Set it to `-1` to have no limit maybe
